src/Common/DNSResolver.cpp

```diff
@@ -289,10 +289,10 @@ void DNSResolver::setDisableCacheFlag(bool is_disabled)
     impl->disable_cache = is_disabled;
 }
 
-void DNSResolver::setCacheMaxSize(const UInt64 cache_max_size)
+void DNSResolver::setCacheMaxEntries(const UInt64 cache_max_entries)
 {
-    impl->cache_address.setMaxSizeInBytes(cache_max_size);
-    impl->cache_host.setMaxSizeInBytes(cache_max_size);
+    impl->cache_address.setMaxSizeInBytes(cache_max_entries);
```

Comment:
Wait a second. Maybe there is a misunderstanding. `setMaxSizeInBytes` sets the maximum total size of the cache. If you like to set the maximum total number of entries, you need to use `setMaxCount`.

Reply:
See discussion above. Default weight policy return 1 on any entry, so setMaxSizeInBytes actually is count